thirdparty-api/data-models.md

```diff
@@ -0,0 +1,953 @@
+- The DFSP requests confirmation from the PISP that the DFSP’s customer has confirmed with the PISP that they authorize the PISP to perform operations on their account. Confirmation requires the PISP to provide the bearer token that the DFSP sent the PISP as confirmation of the successful completion of the out-of-band customer authorization described in the previous step.
+- The DFSP confirms to the PISP the accounts which it will allow the PISP to access and the access types available to the PISP on each account. It also confirms the following items of information:
+  - For each account to which the DFSP grants access, the Mojaloop identifier which the PISP should use in subsequent access requests to identify the account on which the operation should be performed.
+- For each association to be made, the PISP asks the user’s handset to register a keypair to be used to confirm transfer requests in the future. The public key belonging to this keypair is returned to the DFSP, together with the account identifier provided by the DFSP.
+- If the DFSP is not using a local authentication service to verify the challenges it uses to authenticate transfer requests, it asks the scheme’s authentication service to register the public key and associate it with the account ID it provides.
+
+- For each association to be made, the DFSP provides a challenge to the PISP. The PISP asks the customer to sign the challenge, and returns the signed challenge to the DFSP.
+
+- The DFSP verifies that the signed challenge matches the value that it holds for the association, using either its own authentication service or the scheme authentication service.
+The API is used by the following different types of participant, as follows:
+  1. PISPs.
+  2. DFSPs who offer services to their customer which allow the customer to access their account via one or more PISPs.
+  3. FIDO authorization servers.
+  4. The Mojaloop switch
+
+Each resource in the API definition is accompanied by a definition of the type(s) of participant allowed to access it.
+
+### 3.1 Resources
+
+The PISP API will contain the following resources:
+
+#### 3.1.1 `/accounts`
+
+The `/accounts` resource is used to request information from a DFSP relating to the accounts 
+it holds for a given identifier. The identifier is given to the PISP by the user, and the 
+DFSP returns a set of information about the accounts it is prepared to divulge to the PISP.
+The PISP can then display the names of the accounts to the user, and allow the user to select
+the accounts with which they wish to link via the PISP.
+
+The `/accounts` resource supports the endpoints described below.
+
+##### 3.1.1.1 Requests
+
+This section describes the services that a PISP can request on the /accounts resource.
+
+###### 3.1.1.1.1  `GET /accounts/<ID>`
```

Comment:
No. The idea here is that we are using some opaque identifier that is known to the DFSP, and we don't concern ourselves with Type of the identifier. Since the user already has selected their DFSP, we don't need to do a Party Lookup to determine the DFSP the request should be routed to, we don't need to conform to the FSPIOP /Type/Id(/subId) format. 

Really this allows us to ask the question: "What do you use to log in with your bank?"

From a few lines above:
> where `<ID>` is some identifier a user uses to access their account with the DFSP, such as a phone number, email address, clientId.